hs/src/Reach/Simulator/Core.hs

```diff
@@ -301,12 +312,14 @@ instance Interp DLExpr where
         (V_UInt n, V_Address acc) -> do
           case maybe_dlarg of
             Nothing -> do
-              updateLedger acc 0 (+n)
+              updateLedger simContract nwToken (subtract n)
+              updateLedger acc nwToken (+n)
```

Comment:
I think it is less error-prone to guarantee the ledger is always balanced by having a `transfer` function